Mage/src/main/java/mage/abilities/costs/AlternativeSourceCostsImpl.java

```diff
@@ -20,7 +20,10 @@ public abstract class AlternativeSourceCostsImpl extends StaticAbility implement
 
     protected final AlternativeCost alternativeCost;
     protected final String reminderText;
-    private int zoneChangeCounter = 0;
+    protected final String activationKey;
+    protected static String getActivationKey(String name){
```

Comment:
overridable method used in constructor? make final?

Reply:
It's static, which means it's already final/can't be overwritten. The subclasses define their own static `getActivationKey()` functions with no arguments which each call this static function with their own keyword. Maybe I should use different names for the two functions?

Reply:
Ah you're right, sorry